Instant-Messenger/src/server/server.cpp

```diff
@@ -30,6 +31,20 @@ namespace server {
         this->feAddressBook = new FeAddressBook();
         groupManager = new ServerGroupManager(this->feAddressBook);
 
+        rm_listening_serv_addr.sin_family = AF_INET;
+        rm_listening_serv_addr.sin_addr.s_addr = INADDR_ANY;
+        bzero(&(rm_listening_serv_addr.sin_zero), 8);
+
+        rm_listening_socket_fd = 0;
+    }
+
+    void Server::prepareReplicationManager(int rmNumber) {
+        this->setRmNumber(rmNumber);
+        this->connectToRmServers();
+
+        if(this->getRmNumber() > 0) {
```

Comment:
?

Reply:
First RM server should not create listener sockets.